pkg/cel/environment.go

```diff
@@ -54,6 +54,7 @@ func DefaultEnvironment(options ...EnvOption) (*cel.Env, error) {
 	declarations := []cel.EnvOption{
 		ext.Lists(),
 		ext.Strings(),
+		cel.OptionalTypes(),
```

Comment:
Just FYI that Kubernetes has a well defined and versioned, environment that includes options settings and libraries:

xref: https://github.com/kubernetes/kubernetes/blob/9549613361f1c6f82b133a24f9badc70d8caff9d/staging/src/k8s.io/apiserver/pkg/cel/environment/base.go#L232, https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/cel/environment/base.go#L62-L186

Reply:
While in general we can take a look at this, ultimately KRO has opinionation on the CEL syntax it supports so I would hope that we have a controlled subset (not all options) to enable our lib. 

For now I didn't feel the need to extend this but maybe using the k8s version from upstream is eventually beneficial? Im torn on this, happy to hear other thoughts